src/operators/jordanmpotensor.jl

```diff
@@ -0,0 +1,345 @@
+                         B::SparseBlockTensorMap{TB,E,S,2,1},
+                         C::SparseBlockTensorMap{TC,E,S,1,2},
+                         D::SparseBlockTensorMap{TD,E,S,1,1}) where {E,S,TA,TB,TC,TD}
+    allVs = eachspace(V)
+    VA = space(allVs[2:(end - 1), 1, 1, 2:(end - 1)])
+    VA == space(A) || throw(SpaceMismatch())
+
+    VB = removeunit(space(allVs[2:(end - 1), 1, 1, end]), 4)
+    VB == space(B) || throw(SpaceMismatch())
+
+    VC = removeunit(space(allVs[1, 1, 1, 2:(end - 1)]), 1)
+    VC == space(C) || throw(SpaceMismatch())
+
+    VD = removeunit(removeunit(space(allVs[1, 1, 1, end:end]), 4), 1)
+    VD == space(D) || throw(SpaceMismatch())
+
+    return JordanMPOTensor{E,S,TA,TB,TC,TD}(V, A, B, C, D)
+end
+
+function JordanMPOTensor(W::SparseBlockTensorMap{TT,E,S,2,2}) where {TT,E,S}
+    @assert W[1, 1, 1, 1] isa BraidingTensor && W[end, 1, 1, end] isa BraidingTensor
+    # @assert all(I -> I[1] ≤ I[4], nonzero_keys(W))
+
+    A = W[2:(end - 1), 1, 1, 2:(end - 1)]
+    B = W[2:(end - 1), 1, 1, end]
+    C = W[1, 1, 1, 2:(end - 1)]
+    D = W[1, 1, 1, end:end] # ensure still blocktensor to allow for sparse
+
+    return JordanMPOTensor(space(W),
+                           A,
+                           removeunit(B, 4),
+                           removeunit(C, 1),
+                           removeunit(removeunit(D, 4), 1))
+end
+
+function jordanmpotensortype(::Type{S}, ::Type{E}) where {S<:VectorSpace,E<:Number}
+    TA = Union{tensormaptype(S, 2, 2, E),BraidingTensor{E,S}}
+    TB = tensormaptype(S, 2, 1, E)
+    TC = tensormaptype(S, 1, 2, E)
+    TD = tensormaptype(S, 1, 1, E)
+    return JordanMPOTensor{E,S,TA,TB,TC,TD}
+end
+function jordanmpotensortype(::Type{O}) where {O<:MPOTensor}
+    return jordanmpotensortype(spacetype(O), scalartype(O))
+end
+
+function Base.similar(W::JordanMPOTensor, ::Type{T}) where {T<:Number}
+    return JordanMPOTensor{T}(undef, space(W))
+end
+
+# Properties
+# ----------
+TensorKit.space(W::JordanMPOTensor) = W.V
+Base.eltype(::Type{JordanMPOTensor{E,S,TA,TB,TC,TD}}) where {E,S,TA,TB,TC,TD} = TA
+
+function Base.haskey(W::JordanMPOTensor, I::CartesianIndex{4})
+    Base.checkbounds(W, I.I...)
+    # only has braiding tensors if sizes are large enough
+    sz = size(W)
+    (sz[1] > 1 && I == CartesianIndex(1, 1, 1, 1) ||
+     sz[4] > 1 && I == CartesianIndex(sz[1], 1, 1, sz[4])) && return true
+
+    row, col = I.I[1], I.I[4]
+
+    if row == 1 && col == sz[4]
+        return haskey(W.D, CartesianIndex(1, 1))
+    elseif row == 1
+        return haskey(W.C, CartesianIndex(1, 1, col - 1))
+    elseif col == sz[4]
+        return haskey(W.B, CartesianIndex(row - 1, 1, 1))
+    elseif 1 < row < sz[1] && 1 < col < sz[4]
+        return haskey(W.A, CartesianIndex(row - 1, 1, 1, col - 1))
+    else
+        return false
+    end
+end
+
+# TODO: avoid this slow fallback wherever possible:
+Base.parent(W::JordanMPOTensor) = parent(SparseBlockTensorMap(W))
```

Comment:
Where would this be used now?

Reply:
Many of the `AbstractBlockTensorMap` implementations that are "array-like" call out to this function. I've tried to avoid most of them by overloading faster fallbacks, but sometimes this is more or less the most efficient way. I'll just delete the TODO for now, should not be an issue if it doesn't show up in the profiler.